src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -725,7 +725,10 @@ public function update_item( $request ) {
 			);
 		}
 
-		$owner_id = email_exists( $request['email'] );
+		$owner_id = false;
+		if ( is_string( $request['email'] ) ) {
+			$owner_id = email_exists( $request['email'] );
+		}
```

Comment:
Yes, I agree with this change.

* `email_exists()` returns `false` when the email doesn't exist. Therefore, setting `$owner_id` to `false` intentionally retains that value without introducing a BC break.

* If `$request['email']` is not a `string`, then the filter `'email_exists'` will not run.
Technically, not firing the filter is a difference in how the code executes. But, there's no reason to verify a non-string email, as it is not an email.

This code also provides a tiny micro-optimization when the `$request['email']` is a non-string. How?
* `email_exists()` is not invoked
* which means `get_user_by()` is not invoked
* which means `WP_User::get_data_by()` is not invoked
* which means `wp_cache_get()` is not invoked
* which means `'email_exists'` filter is not fired

Reply:
Committed via https://core.trac.wordpress.org/changeset/54317 ✅

Reply:
FYI: This fix reduced the PHPUnit errors from `91` to `75` 🎉 

* [The 8.1 job before the changeset](https://github.com/WordPress/wordpress-develop/actions/runs/3131268429/jobs/5082432524)
* [The 8.1 job after the changeset](https://github.com/WordPress/wordpress-develop/actions/runs/3131279369/jobs/5082454356)